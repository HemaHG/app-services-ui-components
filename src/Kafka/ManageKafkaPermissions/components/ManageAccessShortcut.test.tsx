import { composeStories } from "@storybook/testing-react";
import * as stories from "./ManageAccessShortcut.stories";
import { render, waitForI18n } from "../../../test-utils";

const { MultipleRowsExist, OnlyRowInTheTable } = composeStories(stories);

describe("Manage access shortcut permissions", () => {
  it("should render a permissions table without header with initial state when there are multiple rows in the table", async () => {
    const comp = render(<MultipleRowsExist />);

    await waitForI18n(comp);
    expect(comp.baseElement).toMatchSnapshot();
  });

  it("should render a permissions table and header with initial state when there is only one row in the table", async () => {
    const comp = render(<OnlyRowInTheTable />);

    await waitForI18n(comp);
    expect(comp.baseElement).toMatchSnapshot();
  });
});
