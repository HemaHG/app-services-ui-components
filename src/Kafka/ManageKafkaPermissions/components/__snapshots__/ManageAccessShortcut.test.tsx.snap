// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Manage access shortcut permissions should render a permissions table and header with initial state when there is only one row in the table 1`] = `
<body>
  <div>
    <form
      class="pf-c-form"
      novalidate=""
      style=""
    >
      <table
        class="pf-c-table pf-m-grid-md pf-m-compact"
        data-ouia-component-id="OUIA-Generated-Table-2"
        data-ouia-component-type="PF4/Table"
        data-ouia-safe="true"
        role="grid"
      >
        <thead
          class="pf-m-nowrap"
        >
          <tr
            class=""
            data-ouia-component-id="OUIA-Generated-TableRow-3"
            data-ouia-component-type="PF4/TableRow"
            data-ouia-safe="true"
          >
            <th
              class=""
              scope="col"
            >
              Resource
            </th>
            <th
              class=""
            />
            <th
              class=""
            />
            <th
              class=""
              scope="col"
            >
              Permission
            </th>
          </tr>
        </thead>
        <tbody
          class=""
          role="rowgroup"
        >
          <tr
            class=""
            data-ouia-component-id="OUIA-Generated-TableRow-4"
            data-ouia-component-type="PF4/TableRow"
            data-ouia-safe="true"
          >
            <td
              class=""
            >
              <div
                class="pf-l-flex"
              >
                <div
                  class=""
                >
                  <h6
                    class="pf-c-title pf-m-md"
                    data-ouia-component-id="OUIA-Generated-Title-2"
                    data-ouia-component-type="PF4/Title"
                    data-ouia-safe="true"
                  >
                    Manage access
                  </h6>
                </div>
                <div
                  class=""
                >
                  <svg
                    aria-hidden="true"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    style="vertical-align: -0.125em;"
                    viewBox="0 0 512 512"
                    width="1em"
                  >
                    <path
                      d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 448c-110.532 0-200-89.431-200-200 0-110.495 89.472-200 200-200 110.491 0 200 89.471 200 200 0 110.53-89.431 200-200 200zm107.244-255.2c0 67.052-72.421 68.084-72.421 92.863V300c0 6.627-5.373 12-12 12h-45.647c-6.627 0-12-5.373-12-12v-8.659c0-35.745 27.1-50.034 47.579-61.516 17.561-9.845 28.324-16.541 28.324-29.579 0-17.246-21.999-28.693-39.784-28.693-23.189 0-33.894 10.977-48.942 29.969-4.057 5.12-11.46 6.071-16.666 2.124l-27.824-21.098c-5.107-3.872-6.251-11.066-2.644-16.363C184.846 131.491 214.94 112 261.794 112c49.071 0 101.45 38.304 101.45 88.8zM298 368c0 23.159-18.841 42-42 42s-42-18.841-42-42 18.841-42 42-42 42 18.841 42 42z"
                    />
                  </svg>
                </div>
              </div>
            </td>
            <td
              class=""
            />
            <td
              class=""
            />
            <td
              class=""
            />
            <td
              class=""
            >
              <div
                class="pf-l-flex"
              >
                <div
                  class=""
                >
                  <button
                    aria-disabled="false"
                    class="pf-c-button pf-m-link"
                    data-ouia-component-id="OUIA-Generated-Button-link-2"
                    data-ouia-component-type="PF4/Button"
                    data-ouia-safe="true"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      style="vertical-align: -0.125em;"
                      viewBox="0 0 448 512"
                      width="1em"
                    >
                      <path
                        d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
                      />
                    </svg>
                  </button>
                </div>
              </div>
            </td>
          </tr>
          <tr
            class=""
            data-ouia-component-id="OUIA-Generated-TableRow-5"
            data-ouia-component-type="PF4/TableRow"
            data-ouia-safe="true"
          >
            <td
              class=""
            >
              <span
                class="pf-c-label mas-m-solid"
              >
                <span
                  class="pf-c-label__content"
                >
                  KI
                </span>
              </span>
               
              Kafka instance
               
              story-instance
            </td>
            <td
              class=""
            />
            <td
              class=""
            />
            <td
              class=""
            >
              <div
                class="pf-c-label-group"
              >
                <div
                  class="pf-c-label-group__main"
                >
                  
                  <ul
                    aria-label="Label group category"
                    class="pf-c-label-group__list"
                    role="list"
                  >
                    <li
                      class="pf-c-label-group__list-item"
                    >
                      <span
                        class="pf-c-label pf-m-outline"
                      >
                        <span
                          class="pf-c-label__content"
                        >
                          Allow
                        </span>
                      </span>
                    </li>
                    <li
                      class="pf-c-label-group__list-item"
                    >
                      <span
                        class="pf-c-label pf-m-outline"
                      >
                        <span
                          class="pf-c-label__content"
                        >
                          Alter
                        </span>
                      </span>
                    </li>
                  </ul>
                </div>
              </div>
            </td>
          </tr>
        </tbody>
      </table>
    </form>
  </div>
</body>
`;

exports[`Manage access shortcut permissions should render a permissions table without header with initial state when there are multiple rows in the table 1`] = `
<body>
  <div>
    <form
      class="pf-c-form"
      novalidate=""
      style=""
    >
      <table
        class="pf-c-table pf-m-grid-md pf-m-compact"
        data-ouia-component-id="OUIA-Generated-Table-1"
        data-ouia-component-type="PF4/Table"
        data-ouia-safe="true"
        role="grid"
      >
        <tbody
          class=""
          role="rowgroup"
        >
          <tr
            class=""
            data-ouia-component-id="OUIA-Generated-TableRow-1"
            data-ouia-component-type="PF4/TableRow"
            data-ouia-safe="true"
          >
            <td
              class=""
            >
              <div
                class="pf-l-flex"
              >
                <div
                  class=""
                >
                  <h6
                    class="pf-c-title pf-m-md"
                    data-ouia-component-id="OUIA-Generated-Title-1"
                    data-ouia-component-type="PF4/Title"
                    data-ouia-safe="true"
                  >
                    Manage access
                  </h6>
                </div>
                <div
                  class=""
                >
                  <svg
                    aria-hidden="true"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    style="vertical-align: -0.125em;"
                    viewBox="0 0 512 512"
                    width="1em"
                  >
                    <path
                      d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 448c-110.532 0-200-89.431-200-200 0-110.495 89.472-200 200-200 110.491 0 200 89.471 200 200 0 110.53-89.431 200-200 200zm107.244-255.2c0 67.052-72.421 68.084-72.421 92.863V300c0 6.627-5.373 12-12 12h-45.647c-6.627 0-12-5.373-12-12v-8.659c0-35.745 27.1-50.034 47.579-61.516 17.561-9.845 28.324-16.541 28.324-29.579 0-17.246-21.999-28.693-39.784-28.693-23.189 0-33.894 10.977-48.942 29.969-4.057 5.12-11.46 6.071-16.666 2.124l-27.824-21.098c-5.107-3.872-6.251-11.066-2.644-16.363C184.846 131.491 214.94 112 261.794 112c49.071 0 101.45 38.304 101.45 88.8zM298 368c0 23.159-18.841 42-42 42s-42-18.841-42-42 18.841-42 42-42 42 18.841 42 42z"
                    />
                  </svg>
                </div>
              </div>
            </td>
            <td
              class=""
            />
            <td
              class=""
            />
            <td
              class=""
            />
            <td
              class=""
            >
              <div
                class="pf-l-flex"
              >
                <div
                  class=""
                >
                  <button
                    aria-disabled="false"
                    class="pf-c-button pf-m-link"
                    data-ouia-component-id="OUIA-Generated-Button-link-1"
                    data-ouia-component-type="PF4/Button"
                    data-ouia-safe="true"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      style="vertical-align: -0.125em;"
                      viewBox="0 0 448 512"
                      width="1em"
                    >
                      <path
                        d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
                      />
                    </svg>
                  </button>
                </div>
              </div>
            </td>
          </tr>
          <tr
            class=""
            data-ouia-component-id="OUIA-Generated-TableRow-2"
            data-ouia-component-type="PF4/TableRow"
            data-ouia-safe="true"
          >
            <td
              class=""
            >
              <span
                class="pf-c-label mas-m-solid"
              >
                <span
                  class="pf-c-label__content"
                >
                  KI
                </span>
              </span>
               
              Kafka instance
               
              story-instance
            </td>
            <td
              class=""
            />
            <td
              class=""
            />
            <td
              class=""
            >
              <div
                class="pf-c-label-group"
              >
                <div
                  class="pf-c-label-group__main"
                >
                  
                  <ul
                    aria-label="Label group category"
                    class="pf-c-label-group__list"
                    role="list"
                  >
                    <li
                      class="pf-c-label-group__list-item"
                    >
                      <span
                        class="pf-c-label pf-m-outline"
                      >
                        <span
                          class="pf-c-label__content"
                        >
                          Allow
                        </span>
                      </span>
                    </li>
                    <li
                      class="pf-c-label-group__list-item"
                    >
                      <span
                        class="pf-c-label pf-m-outline"
                      >
                        <span
                          class="pf-c-label__content"
                        >
                          Alter
                        </span>
                      </span>
                    </li>
                  </ul>
                </div>
              </div>
            </td>
          </tr>
        </tbody>
      </table>
    </form>
  </div>
</body>
`;
